### ISSUE TITLE
role argument spec: allow for examples, like with plugins

### ISSUE THREAD

**Issue by evgeni** [ansible/ansible#81734]
### Summary

When writing documentation for plugins, one can use `EXAMPLES` to have a (yaml) block with common examples.

When it comes to role argument specs, there is no comparable feature (arguably, it's not part of the *argument* spec for plugins either…).

It would be cool if we Ansible would allow roles to have examples that can be rendered by `ansible-doc` and `antsibull-doc`

### Issue Type

Feature Idea

### Component Name

role

### Additional Information

```yaml (paste below)

```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

**Comment by ansibot** [comment]
Files identified in the description:

* [`lib/ansible/playbook/role/__init__.py`](https://github.com/ansible/ansible/blob/devel/lib/ansible/playbook/role/__init__.py)

If these files are incorrect, please update the `component name` section of the description or use the [component](https://github.com/ansible/ansibotmini#ansibotmini) bot command.

**Comment by mattclay** [comment]
cc @felixfontein @samccann @newswangerd

**Comment by felixfontein** [comment]
:+1: for adding that.

Depending on the role a more extensive document would also be useful (something with sections and text between examples), but that's a bit too much IMO - ansible-doc shouldn't become an arbitrary documentation rendering engine.

**Comment by samccann** [comment]
@felixfontein a role within a collection could probably do that already with the collection-level docs support we have now. But that doesn't help standalone roles...And I agree- too much for `ansible-doc`.

**Comment by evgeni** [comment]
Can you point me at the collection level docs you're talking about?

**Comment by samccann** [comment]
https://docs.ansible.com/ansible/latest/dev_guide/developing_collections_structure.html#collections-doc-dir  

Notice there is an unfortunate catch-22 - if you want docs like these in docs.ansible.com, you have to write in .rst in a specific directory. If you want docs like these on AutomationHub/GalaxyNG - you have to write in md in a different directory (my guess is it works on galaxyNG but I haven't poked around to verify that).

**Comment by evgeni** [comment]
Ah right, but those would be really at the collection level, not part of the docs shown for a role in a collection (either via ansible-doc or antsibull-docs).

**Comment by samccann** [comment]
correct

**Comment by felixfontein** [comment]
ansible-core: https://github.com/ansible/ansible/pull/82671
ansible-lint: https://github.com/ansible/ansible-lint/pull/4019
antsibull-docs: https://github.com/ansible-community/antsibull-docs/pull/244

**Comment by ssbarnea** [comment]
@evgeni Any updates on this?

**Comment by felixfontein** [comment]
@ssbarnea this is waiting on the Core team, as PRs exists, but the PR for ansible-core needs to be merged first. Right now #82671 has no review and there's no indication whether it even has a chance to get merged eventually.
